frontend/src/metabase/admin/settings/components/ApiKeys/ManageApiKeys.tsx

```diff
@@ -0,0 +1,149 @@
+          refreshList={refreshList}
+        />
+      ) : null}
+      <Stack pl="md" spacing="lg">
+        <Breadcrumbs
+          crumbs={[
+            [t`Authentication`, "/admin/settings/authentication"],
+            [t`API Keys`],
+          ]}
+        />
+        <Group align="end" position="apart">
+          <Stack>
+            <Title>{t`Manage API Keys`}</Title>
+            {!isShowingEmptyTable && (
+              <Text color="text.1">{t`Allow users to use the API keys to authenticate their API calls.`}</Text>
+            )}
+          </Stack>
+          <Button
+            variant="filled"
+            onClick={() => setModal("create")}
+          >{t`Create API Key`}</Button>
+        </Group>
+        <LoadingAndErrorWrapper loading={isLoading} error={error}>
+          <table
+            className="ContentTable border-bottom"
+            data-testid="api-keys-table"
+          >
+            <thead>
+              <tr>
+                <th>{t`Key name`}</th>
+                <th>{t`Group`}</th>
+                <th>{t`Key`}</th>
+                <th>{t`Last Modified By`}</th>
+                <th>{t`Last Modified On`}</th>
+                <th />
+              </tr>
+            </thead>
+            <tbody>
+              {apiKeys.map(apiKey => (
+                <tr key={apiKey.id} className="border-bottom">
+                  <td className="text-bold">{apiKey.name}</td>
+                  <td>{apiKey.group_name}</td>
+                  <td className="text-monospace">{apiKey.masked_key}</td>
+
+                  {/* TODO: replace with creator_name */}
+                  <td>{apiKey.creator_id}</td>
+
+                  <td>{apiKey.updated_at}</td>
+                  <td>
+                    <Group spacing="md">
+                      <Icon
+                        name="pencil"
+                        className="cursor-pointer"
+                        onClick={() => {
+                          setActiveApiKey(apiKey);
+                          setModal("edit");
+                        }}
+                      />
+                      <Icon
+                        name="trash"
+                        className="cursor-pointer"
+                        onClick={() => {
+                          setActiveApiKey(apiKey);
+                          setModal("delete");
+                        }}
+                      />
+                    </Group>
+                  </td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
+          {isShowingEmptyTable && (
+            <Stack
+              h="40rem" // TODO: how to make this fill only available window height?
+              align="center"
+              justify="center"
+              spacing="sm"
+            >
+              <Title>{t`No API keys here yet`}</Title>
+              <Text color="text.1">{t`Create API keys to programmatically authenticate their API calls.`}</Text>
+            </Stack>
```

Comment:
this could also be a nice little <ApiKeysEmptyState /> component

Reply:
599f8a3ed7a2c37af85ab4bb7b0dae78274a9ddc